### PR TITLE
Document mockserver usage for CDK development

### PR DESCRIPTION
**Pull request by maxi297**
## What
This process has been very useful while developing https://github.com/airbytehq/airbyte/issues/29660 as I didn't know what was the exact API where it occurred. Mocking the server allowed me to end-to-end test without bothering on having access to the source

## How
Just documenting the process. Note that there weren't much thought process on which tool to use. Wiremock would have also been very sufficient for this need.

### CODE REVIEW

**Review by marcosmarxm — COMMENTED** [review]
This is quite nice @maxi297 curious to see how this can applied to alpha connectors Airbyte doesn't have access credentials. Also could be a good workflow to request during new sources contributions or to test low-code connectors components :)

**Comment by maxi297** [comment]
We could definitely have this for alpha connectors. I think there could be some kind of priority list like:
* setup with the docker image hosting the actual server with script to populate data (example https://hub.docker.com/r/gitlab/gitlab-ee)
  * benefits: we host the actual API so the behavior should be identical as in production
  * drawback: some data generation is time sensitive or very long to generate
* setup a mock server with expectations as describe in this PR
* given that the mock server is not sufficient, don't test

Note that the mockserver approach is not bullet proof. This relies on some big assumptions like:
* The API behaves as documented (this can probably still be covered by CATs but the subset of tests will be much smaller)
* We can understand the behavior of the API from the documentation (I don’t have a clear vision of a safety net for this today unfortunately)

File: airbyte-cdk/python/README.md [hunk 1]
```diff
@@ -130,6 +130,27 @@ To run acceptance tests for multiple connectors using the local CDK, from the ro
 ./airbyte-cdk/python/bin/run-cats-with-local-cdk.sh -c <connector1>,<connector2>,...
 ```
 
+#### When you don't have access to the API
+There can be some time where you do not have access to the API (either because you don't have the credentials, network access, etc...) You will probably still want to do end-to-end testing at least once. In order to do so, you can emulate the server you would be reaching using a server stubbing tool.
+
+For example, using `mockserver`, you can set up an expectation file like this:
```
Comment (brianjlai) [inline airbyte-cdk/python/README.md:136]: nit, might be worth a link to the documentation site about how to use this tool for more detail

**Review by brianjlai — APPROVED** [review]
nothing blocking but some clarifying questions

**Pull request merged**